ush/config_defaults.yaml

```diff
@@ -2513,8 +2474,8 @@ verification:
   # format.)
   #
   FCST_SUBDIR_TEMPLATE: '{% if user.RUN_ENVIR == "nco" %}${NET_default}.{init?fmt=%Y%m%d?shift=-${time_lag}}/{init?fmt=%H?shift=-${time_lag}}{% else %}{init?fmt=%Y%m%d%H?shift=-${time_lag}}{% if global.DO_ENSEMBLE %}/${ensmem_name}{% endif %}/postprd{% endif %}'
-  FCST_FN_TEMPLATE: '${NET_default}.t{init?fmt=%H?shift=-${time_lag}}z{% if user.RUN_ENVIR == "nco" and global.DO_ENSEMBLE %}.${ensmem_name}{% endif %}.prslev.f{lead?fmt=%HHH?shift=${time_lag}}.${POST_OUTPUT_DOMAIN_NAME}.grib2'
-  FCST_FN_TEMPLATE_PCPCOMBINE_OUTPUT: '${NET_default}.t{init?fmt=%H}z{% if user.RUN_ENVIR == "nco" and global.DO_ENSEMBLE %}.${ensmem_name}{% endif %}.prslev.f{lead?fmt=%HHH}.${POST_OUTPUT_DOMAIN_NAME}_${VAR}_a${ACCUM_HH}h.nc'
+  FCST_FN_TEMPLATE: '${NET_default}.t{init?fmt=%H?shift=-${time_lag}}z{% if user.RUN_ENVIR == "nco" and global.DO_ENSEMBLE %}.${ensmem_name}{% endif %}.prslev.f{lead?fmt=%HHH?shift=${time_lag}}.{{ task_run_post.post_output_domain_name }}.grib2'
+  FCST_FN_TEMPLATE_PCPCOMBINE_OUTPUT: '${NET_default}.t{init?fmt=%H}z{% if user.RUN_ENVIR == "nco" and global.DO_ENSEMBLE %}.${ensmem_name}{% endif %}.prslev.f{lead?fmt=%HHH}.{{ task_run_post.post_output_domain_name }}_${VAR}_a${ACCUM_HH}h.nc'
```

Comment:
Definitely not asking for it here/now, but we could consider DRYing some of these complex and at least partially repetitive values out via intermediate variables. I guess they'd have to live in some other section, but naming them separately might both reduce the complexity of some expressions and have some explanatory value. It's almost like a spreadsheet, where you might have a block somewhere where various values are being calculated, and then other parts of the spreadsheet just reference those common values. Just a maybe-someday thought.

Reply:
I've been having a lot of trouble with this bit. There are 4 languages we may need to support for defining these paths/filenames that contain cycle-specific information.

Here, it's the MET/METPlus convention of setting time like `{init?fmt=%H}`. When we want to reference this file in Rocoto dependencies, for example, we'd do it with a cycle sting that looks like `<cyclestr>@H</cyclestr>`. And, in GSL's pygraf, we use a Python template like `{FCST_TIME:02d}`. In uwtools, we have used the Pythonic way to deal with the datetime object directly `{{ cycle.strftime('%H') }}`. 

I'm not sure how to deal with this to simply DRY the name of the file that comes out of UPP.

Reply:
That sounds like a higher-level DRY question -- I was only thinking of the immediate, in-the-small case of de-duplicating some Jinja2 expressions. I don't know enough to suggest anything for minimizing the number of representations of the same filename. Maybe there's no way around passing through unevaluated non-Jinja2 template syntax so that each tool sees what it likes.

Reply:
I am re-reading your comment above. I strayed a bit with my first response. I 100% think that the `RUN_ENVIR` flag should be factored out in a similar way as the `CPL_AQM` flag with its own YAML. I might even consider removing those options altogether in our first pass to reduce this thing to just running it like a non-NCO user might. Then we could show how it's all down to configuration settings if they need to change certain bits and pieces for operations.